src/main/resources/test_files/runtime-spark/NestedFLWORSpark2.iq

```diff
@@ -0,0 +1,4 @@
+(:JIQS: ShouldRun; Output="1" :)
```

Comment:
I am actually surprised that this one works, because parallelize() should trigger an RDD, I will double-check. It may be that the implementation of the parallelize iterator bypasses the RDD in case of a local evaluation -- which is not necessarily bad but at least we should understand what is going on, and that this is specific to parallelize.

Reply:
This query also works.
jiqs$ for $i in parallelize(1) for $l in (let $k := json-file("./src/main/resources/queries/conf-ex.json") return $k)
> return $l
>>> 
>>> 
{ "guess" : "Latvian", "target" : "Russian", "country" : "AU", "choices" : [ "Lao", "Latvian", "Russian", "Swahili" ], "sample" : "b7df3f9d67cef259fbcaa5abcad9d774", "date" : "2013-08-20" }
{ "guess" : "Russian", "target" : "Russian", "country" : "AU", "choices" : [ "Croatian", "Nepali", "Russian", "Slovenian" ], "sample" : "8a59d48e99e8a1df7e366c4648095e27", "date" : "2013-08-20" }
{ "guess" : "Czech", "target" : "Czech", "country" : "SE", "choices" : [ "Maori", "Czech", "Korean", "Turkish" ], "sample" : "1787b5c79a00b3513ce76847bc1f5b75", "date" : "2013-08-20" }
{ "guess" : "Greek", "target" : "Serbian", "country" : "SE", "choices" : [ "German", "Greek", "Kannada", "Serbian" ], "sample" : "0d5b697ebb326b5043ce7fa60a7b968d", "date" : "2013-08-20" }
{ "guess" : "Serbian", "target" : "Serbian", "country" : "AU", "choices" : [ "Dari", "Serbian", "Sinhalese", "Vietnamese" ], "sample" : "0d5b697ebb326b5043ce7fa60a7b968d", "date" : "2013-08-20" }

Maybe I can be of help in understanding this one. The reason is that we support storing RDDs without materialization in let clauses now, The inner FLWOR expression, that is (let $i := parallelize(1) return 1), does not require a nested DataFrame/UDF operation

Reply:
With our recent changes let clause is evaluated locally. This triggers local evaluation for return clause as well. This local evaluation triggers getLocalValue from DynamicContext for the VariableReferenceIteratorExpression in the body of the return clause.

Since no nesting takes place, the query doesn't crash

Reply:
@CanBerker you are correct of course. Indeed, there is no actual Spark job triggered for $k. This is unexpectedly wonderful, because this expands our possibilities.